runtime/lib/ttnn/program.cpp

```diff
@@ -278,16 +278,26 @@ run(::tt::target::ttnn::TransposeOp const *op, ::ttnn::device::Device &device,
   int32_t dimension1 = op->dimension1();
   int32_t dimension2 = op->dimension2();
   auto input_rank = in.get_shape().rank();
-  std::vector<std::int64_t> dimensionOrder(input_rank);
+  // for the current version of permute, we need to work in 4D, so we add
```

Comment:
Have we filed issues (both for `permute` not handling non-4d tensors, and `transpose` failing to execute) on tt-metal?

Reply:
Perhaps related: https://github.com/tenstorrent/tt-metal/issues/11191

Reply:
Thanks @nsmithtt , we saw that one as well. I will run a small python test using metal to double-check.